### PR TITLE
Update meta.yaml to use the most recent compilers

### PR DESCRIPTION
**Pull request by marcelotrevisani**
Testing the directives of compiler('c') and compiler('cxx') to use the newest compilers

### CODE REVIEW

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please re-render

**Comment by marcelotrevisani** [comment]
@conda-forge/qwt Could someone please review that for me?

**Comment by jschueller** [comment]
looks consistent with the doc, the build number should be bumped

File: .ci_support/win_c_compilervs2015cxx_compilervs2015python3.5.yaml [hunk 1]
```diff
@@ -9,6 +9,6 @@ pin_run_as_build:
 python:
 - '3.5'
 zip_keys:
-- - c_compiler
+- - python
+  - c_compiler
```
Comment (jschueller) [inline .ci_support/win_c_compilervs2015cxx_compilervs2015python3.5.yaml]: did you have to edit these manually ??

Comment (marcelotrevisani) [reply .ci_support/win_c_compilervs2015cxx_compilervs2015python3.5.yaml]: No, I didn't. I just edited the ```meta.yaml``` the other files were the rerender process

**Comment by marcelotrevisani** [comment]
@jschueller It failed on python 35 in Travis but it's not related to this commit. 

```
  File "/Users/travis/miniconda3/lib/python3.6/site-packages/urllib3/connection.py", line 141, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "/Users/travis/miniconda3/lib/python3.6/site-packages/urllib3/util/connection.py", line 83, in create_connection
    raise err
  File "/Users/travis/miniconda3/lib/python3.6/site-packages/urllib3/util/connection.py", line 73, in create_connection
    sock.connect(sa)
OSError: [Errno 51] Network is unreachable
```
Could we merge that or do you prefer to rerun this CI ?

**Comment by jschueller** [comment]
its ok

**Pull request merged**